Core/src/Utilities/AnnealingUtility.cpp

```diff
@@ -8,16 +8,20 @@
 
 #include "Acts/Utilities/AnnealingUtility.hpp"
 
+#include "Acts/Utilities/AlgebraHelpers.hpp"
+
+namespace {
 /// @brief Gaussian-like function for weight calculation
 /// Note: Factor 2 in denominator is included in inverse temperature
 ///
 /// @param chi2 Chi^2 value
 /// @param invTemp Denominator 1/(2 * temperature)
 ///
 /// @return exp(-chi2 * invTemp)
-static double computeAnnealingWeight(double chi2, double invTemp) {
-  return std::exp(-chi2 * invTemp);
+double computeAnnealingWeight(double chi2, double invTemp) {
+  return Acts::safeExp(-chi2 * invTemp);
```

Comment:
Should we think about different cut-off values for doubles in `safeExp` or should we leave this for a later PR?

Reply:
Fundamentally I suppose this also depends on the exact use case, i.e. what kind of calculation we're doing?

Reply:
Hm why would it? 
Depending on how precise one wants to calculate the exponential, one needs to call `safeExp` with `float`, `double`, `long double`, etc. I was just thinking that for each of them we should provide a different cut-off value in `safeExp` (`50` for `float`, some other value for `double`, etc.). 

This might not be necessary, just wanted to mention it.

Reply:
Hmm I tried some things at godbold, and believe we should do separate limits: https://godbolt.org/z/z53Er6Mzf

I would propose:
```c++
template<typename T> struct ExpSafeLimit {};
template<> struct ExpSafeLimit<double> { constexpr static double value = 700.0; };
template<> struct ExpSafeLimit<float> { constexpr static float value = 85.0; };
```